system/lib/pthread/pthread_create.c

```diff
@@ -51,8 +53,32 @@ void __do_cleanup_pop(struct __ptcb *cb) {
   __pthread_self()->cancelbuf = cb->__next;
 }
 
-int __pthread_create(pthread_t *thread, const pthread_attr_t *attr, void *(*start_routine) (void *), void *arg) {
-  return __pthread_create_js(thread, attr, start_routine, arg);
+int __pthread_create(pthread_t *restrict res, const pthread_attr_t *restrict attrp, void *(*entry)(void *), void *restrict arg) {
+  if (!res) {
+    return EINVAL;
+  }
+
+  // Allocate thread block (pthread_t structure).
+  struct pthread *new = malloc(sizeof(struct pthread));
```

Comment:
Note to myself: use `emscripten_builtin_malloc` here to prevent LSan from considering it as leak if we remove those `__lsan_{disable,enable}()` calls in the future. Probably PR #14481 needs to land first (to ensure the uninstrumented `emscripten_builtin_free` is called in `freeThreadData`).

Reply:
Hmm.. this code isn't wrapped in `__lsan_{disable,enable}()` so I wonder why I don't need `emscripten_builtin_malloc` today?

Ah, I see, that `__lsan_{disable,enable}()` is only needed for the proxied call.   This code runs on the thread calling `pthread_create` which is guess is already wrapped by lsan itself (this is why we have `emscripten_builtin_pthread_create`).

Reply:
Ah I see. The `emscripten_builtin_pthread_create` -> `__pthread_create_js` change in `_do_call` doesn't end up calling the internal `__pthread_create` function, so this isn't necessary at all, interesting.